app_dart/lib/src/request_handlers/dart_internal_subscription.dart

```diff
@@ -49,18 +49,30 @@ class DartInternalSubscription extends SubscriptionHandler {
     //
     // Example: https://flutter.googlesource.com/recipes/+/c6af020f0f22e392e30b769a5ed97fadace308fa/recipes/engine_v2/engine_v2.py#185
     log.info("Getting buildbucket id from pubsub message");
-    final int buildbucketId = int.parse(json.decode(message.data.toString())['buildbucket_id']);
+    final dynamic messageJson = json.decode(message.data.toString());
+
+    final int buildbucketId = messageJson['buildbucket_id'];
+    log.info("Buildbucket id: $buildbucketId");
 
     log.info("Creating build request object");
-    final GetBuildRequest request = GetBuildRequest(id: buildbucketId.toString());
+    final GetBuildRequest request = GetBuildRequest(
+        id: messageJson['buildbucket_id'].toString(),
+        fields:
+            "id,builder,number,createdBy,createTime,startTime,endTime,updateTime,status,input.properties,input.gitilesCommit");
 
     log.info(
       "Calling buildbucket api to get build data for build $buildbucketId",
     );
     final Build build = await _getBuildFromBuildbucket(request);
 
+    String? name;
+    if (build.input?.properties != null && build.input?.properties?["build"] != null) {
+      final Map<String, Object> buildProperties = build.input?.properties?["build"] as Map<String, Object>;
+      name = buildProperties["name"] as String;
```

Comment:
Is this property available in builds from both engine and flutter?

Reply:
Yes, [here's an example from the packaging_builder run on the drone.](https://ci.chromium.org/p/dart-internal/builders/flutter/Linux%20Production%20Engine%20Drone/19391?)